tools/hangar/__snapshots__/test_corpus/valid/struct_from_json.w_compile_tf-aws.md

```diff
@@ -0,0 +1,761 @@
+      const errors = validator.validate(obj, this._schema).errors;
+      if (errors.length > 0) {
+        throw new Error(`unable to parse ${this.name}:\n ${errors.join("\n- ")}`);
+      }
+      return obj;
+    }
+    static _toInflightType(context) {
+      return fromInline(`require("./Person.Struct.js")(${ context._lift(stdStruct) })`);
+    }
+  }
+  return Person;
+};
+
+```
+
+## Student.Struct.js
+```js
+module.exports = function(stdStruct, fromInline) {
+  class Student {
+    static _schema = {
+      id: "/Student",
+      type: "object",
+      properties: {
+        ...require("./Person.Struct.js")()._schema.properties,
+        enrolled: { type: "boolean" },
+        schoolId: { type: "string" },
+        advisor: { "$ref": "/Advisor" },
+        enrolledCourses: { type: "array", uniqueItems: true, items: { "$ref": "/Course" } },
+        coursesTaken: { type: "array",  items: { "$ref": "/CourseResults" } },
+        additionalData: { type: "object" },
+      },
+      required: [
+        "enrolled",
+        "schoolId",
+        ...require("./Person.Struct.js")()._schema.required,
+      ]
+    }
+    static _getDependencies() {
+      return {
+        "/Advisor": require("./Advisor.Struct.js")()._schema,
+        ...require("./Advisor.Struct.js")()._getDependencies(),
+        "/Course": require("./Course.Struct.js")()._schema,
+        ...require("./Course.Struct.js")()._getDependencies(),
+        "/CourseResults": require("./CourseResults.Struct.js")()._schema,
+        ...require("./CourseResults.Struct.js")()._getDependencies(),
+      }
+    }
+    static _validate(obj) {
+      const validator = stdStruct._getValidator(this._getDependencies());
+      const errors = validator.validate(obj, this._schema).errors;
+      if (errors.length > 0) {
+        throw new Error(`unable to parse ${this.name}:\n ${errors.join("\n- ")}`);
+      }
+      return obj;
+    }
+    static _toInflightType(context) {
+      return fromInline(`require("./Student.Struct.js")(${ context._lift(stdStruct) })`);
+    }
+  }
+  return Student;
+};
+
+```
+
+## inflight.$Closure1.js
+```js
+module.exports = function({ $Student }) {
+  class $Closure1 {
+    constructor({  }) {
+      const $obj = (...args) => this.handle(...args);
+      Object.setPrototypeOf($obj, this);
+      return $obj;
+    }
+    async handle() {
+      const jStudent3 = Object.freeze({"firstName":"struct","lastName":"greatest","enrolled":true,"schoolId":"s3-inflight","dob":Object.freeze({"month":4,"day":1,"year":1999}),"coursesTaken":[Object.freeze({"grade":"B","dateTaken":Object.freeze({"month":5,"day":10,"year":2021}),"course":Object.freeze({"name":"COMP 101","credits":2})}), Object.freeze({"grade":"A","dateTaken":Object.freeze({"month":5,"day":10,"year":2021}),"course":Object.freeze({"name":"COMP 121","credits":4})})]});
+      const studentInflight1 = ((j, s) => {return s._validate(j)})(jStudent3, $Student);
```

Comment:
Curious why isn't this just calling a method `MyStruct.fromJson()`?

What's the motivation for all the macros?

Reply:
It does now 😁